src/main/java/org/esbtools/auth/jboss/CertLdapLoginModule.java

```diff
@@ -183,11 +189,25 @@ public void initializeRolesProvider() throws Exception {
     }
 
     private void validateEnvironment(String certificatePrincipal) throws NamingException {
-        if(StringUtils.isNotBlank(environment)) {
-            String location = getLDAPAttribute(certificatePrincipal, LOCATION);
 
-            if(!StringUtils.equals(environment, location)) {
-                throw new NoSuchAttributeException("Location from certificate does not match configured environment");
+        String ou = getLDAPAttribute(certificatePrincipal, OU);
+        String location = getLDAPAttribute(certificatePrincipal, LOCATION);
+
+        if(StringUtils.isBlank(ou)) {
+            throw new NoSuchAttributeException("No ou in dn, you may need to update your certificate: " + certificatePrincipal);
+        } else {
+            if(ALL_ACCESS_OU.equalsIgnoreCase(StringUtils.replace(ou, " ", ""))){
+                LOGGER.debug("Skipping environment validation, user ou matches {} ", ALL_ACCESS_OU);
+            } else {
+                //if dn not from allAccessOu, verify the location (l) field
+                //in the cert matches the configured environment
+                if(StringUtils.isBlank(location)) {
+                    throw new NoSuchAttributeException("No location in dn, you may need to update your certificate: " + certificatePrincipal);
+                } else if (StringUtils.isBlank(ou)){
```

Comment:
The proceeding if(StringUtils.isBlank(ou)) { ... } is exactly this check, is it not? Should this be some other check? Or just removed?

Reply:
Yeah this can actually be removed I think